### PR TITLE
MTDSA-12939: update AmendPensionsReliefs to use tys endpoint & update…

### PR DESCRIPTION
**Pull request by QHMRC**
… unit/Inter tests

### CODE REVIEW

File: app/v1/connectors/AmendPensionsReliefsConnector.scala [hunk 1]
```diff
@@ -29,12 +30,21 @@ import scala.concurrent.{ExecutionContext, Future}
...
+  def createOrAmendPensionsRelief(request: AmendPensionsReliefsRequest)(implicit
+      hc: HeaderCarrier,
+      ec: ExecutionContext,
+      correlationId: String): Future[DownstreamOutcome[Unit]] = {
+
+    val downstreamUri =
+      if (request.taxYear.useTaxYearSpecificApi) {
+        TaxYearSpecificIfsUri[Unit](s"income-tax/reliefs/pensions/${request.taxYear.asTysDownstream}/${request.nino.nino}")
+      } else {
+        DesUri[Unit](s"income-tax/reliefs/pensions/${request.nino.nino}/${request.taxYear.asMtd}")
```
Comment (harry-ritchie) [inline app/v1/connectors/AmendPensionsReliefsConnector.scala:42]: The format of taxYear '.asTysDownstream' and '.asMtd' is correct here, but the comments for '.asMtd' (in TaxYear.scala) has the wrong format. I did change this myself but seems to still be wrong. Could you change these comments again please.

File: app/v1/controllers/AmendPensionsReliefsController.scala [hunk 2]
```diff
@@ -110,7 +110,7 @@ class AmendPensionsReliefsController @Inject() (val authService: EnrolmentsAuthS
           RuleTaxYearNotSupportedError | MtdErrorWithCustomMessage(ValueFormatError.code) =>
         BadRequest(Json.toJson(errorWrapper))
       case InternalError => InternalServerError(Json.toJson(errorWrapper))
-      case _               => unhandledError(errorWrapper)
+      case _             => unhandledError(errorWrapper)
```
Comment (harry-ritchie) [inline app/v1/controllers/AmendPensionsReliefsController.scala:113]: The format of this 'errorResult' method needs to be in the same format of the CreateAmendUkDividendsAnnualSummaryController.errorResult. 

With the 'MtdErrorWithCustomMessage(ValueFormatError.code)', can we not just write this as 'ValueFormatError'?

File: it/v1/endpoints/AmendPensionsReliefsControllerISpec.scala [hunk 3]
```diff
@@ -53,27 +65,27 @@ class AmendPensionsReliefsControllerISpec extends IntegrationBaseSpec {
...
          |    }
          |  ]
          |}
          |""".stripMargin)
 
-    def uri: String = s"/pensions/$nino/$taxYear"
+    def uri: String = s"/pensions/$nino/$mtdTaxYear"
 
-    def desUri: String = s"/income-tax/reliefs/pensions/$nino/$taxYear"
+    def downstreamUri: String
```
Comment (harry-ritchie) [inline it/v1/endpoints/AmendPensionsReliefsControllerISpec.scala:88]: Could maybe move this (def downstreamUri) above to line 47 just to group the abstract methods together?

File: it/v1/endpoints/AmendPensionsReliefsControllerISpec.scala [hunk 4]
```diff
@@ -241,11 +268,14 @@ class AmendPensionsReliefsControllerISpec extends IntegrationBaseSpec {
           (BAD_REQUEST, "INVALID_TAXABLE_ENTITY_ID", BAD_REQUEST, NinoFormatError),
           (BAD_REQUEST, "INVALID_TAX_YEAR", BAD_REQUEST, TaxYearFormatError),
           (BAD_REQUEST, "INVALID_PAYLOAD", INTERNAL_SERVER_ERROR, InternalError),
+          (BAD_REQUEST, "INVALID_CORRELATIONID", INTERNAL_SERVER_ERROR, InternalError),
```
Comment (harry-ritchie) [inline it/v1/endpoints/AmendPensionsReliefsControllerISpec.scala:271]: I think we should also split this up to have two sequence variables (errors + extraTysErrors) and concatenate them together like we did for the service.

File: test/v1/services/AmendPensionsReliefsServiceSpec.scala [hunk 5]
```diff
@@ -83,11 +92,13 @@ class AmendPensionsReliefsServiceSpec extends UnitSpec {
         }
 
       val input = Seq(
+        "INVALID_CORRELATIONID"     -> InternalError,
```
Comment (harry-ritchie) [inline test/v1/services/AmendPensionsReliefsServiceSpec.scala:95]: Again, I think we can split this up into val errors and val extraTysErrors and concatenate them together

File: app/v1/controllers/AmendPensionsReliefsController.scala (hunk 2 above)
Comment (QHMRC) [reply app/v1/controllers/AmendPensionsReliefsController.scala]: made the changes to the errorResult method @harry-ritchie

File: app/v1/connectors/AmendPensionsReliefsConnector.scala (hunk 1 above)
Comment (QHMRC) [reply app/v1/connectors/AmendPensionsReliefsConnector.scala:42]: did a rebase

**Review by harry-ritchie — APPROVED** [review]

**Pull request merged**